### PR TITLE
C# Rewrite

### LINKED ISSUE #70: Existing embedded artwork not loading
When I open an album or song with existing embedded artwork (which shows up in music players on multiple operating systems, so I know it exists), it doesn't render. Selecting every song in an album or just a single song both don't show the artwork. If I use the file picker and upload new album art, that does render. This makes it difficult to know if an album already has existing artwork. 

OS: Fedora 37
GNOME Version: 43.1

### LINKED ISSUE #74: Double digit track number
I usually use double digit track numbers when tagging, to ensure proper filename sorting when renaming files from tags.
Even if I insert a 0-padded double digit track number (e.g. `09`), Tagger saves it as a single digit (`9`).

Any chance this behavior could be changed?

Thanks for the great program btw

### LINKED ISSUE #79: C# Rewrite
Same as Denaro and Tube Converter

### LINKED ISSUE #85: Pressing Enter in the convert filename to tag popup activates Cancel
Steps to reproduce:

1. select an mp3 file with a filename Artist - Title.mp3 and with no ID3 tag
2. press Ctrl+F to Convert Filename to Tag
3. press Enter

Expectation:

- the Ok button (highlighted in blue) gets triggered and empty artist/title metadata fields get filled in

Actual results:

- the Cancel button gets triggered and the popup is dismissed without filling anything in

Version: 2022.11.2 from Flathub

### LINKED ISSUE #86: Music file album art deleted downloading metadata
Great app.
I think it could be useful a separate toggle to decide if the album art is deleted when downloading the metadata but there isn't an album cover on the website.
Because sometimes the name, artist... are wrong, but present and there is an album cover and if I disable "Overwite tag with musicbrainz" nothing is written

### PR DESCRIPTION
**Pull request by nlogozzo**
Closes #70 
Closes #74 
Closes #79 
Closes #85 
Closes #86

### CODE REVIEW

File: NickvisionTagger.Shared/Helpers/MediaHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    
+    /// <summary>
+    /// Converts a file size in bytes to a string (~ MB)
+    /// </summary>
+    /// <param name="fileSize">The file size in bytes</param>
+    /// <returns>The file size string (~ MB)</returns>
+    public static string ToFileSizeString(long fileSize)
+    {
+        var sizes = new string[] { _("B"), _("KB"), _("MB"), _("GB"), _("TB") };
```
[the remark this answers is not in the document]
Comment (nlogozzo) [reply NickvisionTagger.Shared/Helpers/MediaHelpers.cs:33]: Does `B` stay `B` or is there like `iB`?

[the remark this answers is not in the document]
Comment (fsobolev) [reply NickvisionTagger.Shared/Helpers/MediaHelpers.cs]: It stays `B`

[the remark this answers is not in the document]
Comment (nlogozzo) [reply NickvisionTagger.Shared/Helpers/MediaHelpers.cs]: That's what I figured :)

File: flatpak/org.nickvision.tagger.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    },
+    "command" : "/app/bin/org.nickvision.tagger",
+    "finish-args":[
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=ipc",
+        "--share=network",
+        "--filesystem=home"
```
Comment (DaPigGuy) [inline flatpak/org.nickvision.tagger.json:23]: Any reason for this and not portals?

Comment (nlogozzo) [reply flatpak/org.nickvision.tagger.json:23]: Not in particular...I remember portals us issues with folders as compared to files...as with folder we weren't gettong the correct path returns, or it was still read-only and users couldn't tag any files.

**Comment by nlogozzo** [comment]
@soumyaDghosh or @fsobolev could you update the `snapcraft.yml` file?

**Pull request merged**